### PR TITLE
vttablet-up.sh : Adds detection of mysqld launch failure.

### PR DESCRIPTION
**Pull request by chrisrgithub**
Given failure, tests for further for presence of apparmor.  Exits with error and recommendation to disable.

### CODE REVIEW

**Comment by morgo** [comment]
I think we should try and embed this check in mysqlctl itself. Otherwise, users who don't use our examples won't benefit.

**Comment by deepthi** [comment]
> I think we should try and embed this check in mysqlctl itself. Otherwise, users who don't use our examples won't benefit.

Good point, but worth doing in the example until that is implemented.

File: examples/local/vttablet-up.sh [hunk 1]
```diff
@@ -71,11 +76,27 @@ for uid_index in $uids; do
...
     -log_dir $VTDATAROOT/tmp \
     -tablet_uid $uid \
     -mysql_port $mysql_port \
-    $action &
+    $action 
+
+    err=$?
+    if [[ $err -ne 0 ]]; then
+	echo "Checking for apparmor status for mysqld:"
+	sudo aa-status | grep mysqld
```
Comment (morgo) [inline examples/local/vttablet-up.sh]: Stock Ubuntu will pause and wait for a password here. It is only cloud instances that have sudo no password configured. We should instead recommend running this command to check.

File: examples/local/vttablet-up.sh [hunk 2]
```diff
@@ -27,6 +27,11 @@ grpc_port_base=$[16000 + $uid_base]
 mysql_port_base=$[17000 + $uid_base]
 tablet_hostname=''
 
+function fail() {
+    echo "ERROR: $1"
+    exit 1
+}
+
 # Travis hostnames are too long for MySQL, so we use IP.
```
Comment (morgo) [inline examples/local/vttablet-up.sh]: This function should already be present, because it is in env.sh

File: examples/local/vttablet-up.sh (hunk 1 above)
Comment (deepthi) [reply examples/local/vttablet-up.sh]: That is right.. @chrisrgithub you will need to remove this check but still print a helpful message.

**Comment by chrisrgithub** [comment]
- Updated for passive check only, with more info on ways to check for apparmor or selinux protection.
- Preserved Morgo's more recent changes.
- Note that if the user simply "tries again", even after disabling apparmor it will hang silently while processing the existing vtdataroot.  Hence the advice to empty vtdatarot first before retrying.

**Pull request merged**